tsl/src/data_node.c

```diff
@@ -589,8 +593,29 @@ data_node_validate_extension_availability(TSConnection *conn)
 				 errmsg("TimescaleDB extension not available on remote PostgreSQL instance"),
 				 errhint("Install the TimescaleDB extension on the remote PostgresSQL instance.")));
 
-	/* Here we validate the available version, not the installed version */
-	remote_validate_extension_version(conn, PQgetvalue(res, 0, 0));
+	Assert(PQnfields(res) == 1);
+
+	int i;
+	StringInfo concat_versions = makeStringInfo();
+	bool old_version;
+	for (i = 0; i < PQntuples(res); i++)
+	{
+		appendStringInfo(concat_versions, "%s, ", PQgetvalue(res, i, 0));
+		compatible = dist_util_is_compatible_version(PQgetvalue(res, i, 0),
+													 TIMESCALEDB_VERSION,
+													 &old_version);
+		if (compatible)
```

Comment:
Isn't this version should be installed in such case during the `CREATE EXTENSION` operation on data node? I don't think we specify any version there

Reply:
Right, we install the `default` version and that default might not always be compatible with the one on AN (eg. default is 1.7.4 and we have 2.0.0 as well). But in that case a user can ALTER EXTENSION UPDATE on DN and the problem will be solved.